Dockerfile

```diff
@@ -101,46 +93,15 @@ FROM build-deps AS repaired
 
 COPY ./ /pyheif
 
-# python 3.6
-RUN set -ex \
-    && cd "/opt/python/cp36-cp36m/bin/" \
-    && ./pip wheel /pyheif \
-    && auditwheel repair pyheif*.whl --plat manylinux2014_x86_64 -w /wheelhouse
-# python 3.7
-RUN set -ex \
-    && cd "/opt/python/cp37-cp37m/bin/" \
-    && ./pip wheel /pyheif \
-    && auditwheel repair pyheif*.whl --plat manylinux2014_x86_64 -w /wheelhouse
-# python 3.8
-RUN set -ex \
-    && cd "/opt/python/cp38-cp38/bin/" \
-    && ./pip wheel /pyheif \
-    && auditwheel repair pyheif*.whl --plat manylinux2014_x86_64 -w /wheelhouse
-# python 3.9
-RUN set -ex \
-    && cd "/opt/python/cp39-cp39/bin/" \
-    && ./pip wheel /pyheif \
-    && auditwheel repair pyheif*.whl --plat manylinux2014_x86_64 -w /wheelhouse
-# python 3.10
-RUN set -ex \
-    && cd "/opt/python/cp310-cp310/bin/" \
-    && ./pip wheel /pyheif \
-    && auditwheel repair pyheif*.whl --plat manylinux2014_x86_64 -w /wheelhouse
-# python 3.11
-RUN set -ex \
-    && cd "/opt/python/cp311-cp311/bin/" \
-    && ./pip wheel /pyheif \
-    && auditwheel repair pyheif*.whl --plat manylinux2014_x86_64 -w /wheelhouse
-# pypy 3.7
-RUN set -ex \
-    && cd "/opt/python/pp37-pypy37_pp73/bin/" \
-    && ./pip wheel /pyheif \
-    && auditwheel repair pyheif*.whl --plat manylinux2014_x86_64 -w /wheelhouse
-# pypy 3.8
-RUN set -ex \
-    && cd "/opt/python/pp38-pypy38_pp73/bin/" \
-    && ./pip wheel /pyheif \
-    && auditwheel repair pyheif*.whl --plat manylinux2014_x86_64 -w /wheelhouse
+RUN /opt/python/cp37-cp37m/bin/pip wheel /pyheif
+RUN /opt/python/cp38-cp38/bin/pip wheel /pyheif
+RUN /opt/python/cp39-cp39/bin/pip wheel /pyheif
+RUN /opt/python/cp310-cp310/bin/pip wheel /pyheif
+RUN /opt/python/cp311-cp311/bin/pip wheel /pyheif
+RUN /opt/python/pp37-pypy37_pp73/bin/pip wheel /pyheif
+RUN /opt/python/pp38-pypy38_pp73/bin/pip wheel /pyheif
+RUN /opt/python/pp39-pypy39_pp73/bin/pip wheel /pyheif
+RUN auditwheel repair pyheif*.whl --plat $PLAT -w /wheelhouse
```

Comment:
Use `$PLAT` argument here

Reply:
Because you're using the PLAT arg here, ARG PLAT needs to be defined in this scope. Docker doesn't seem to care but other builders do as specified here: https://docs.docker.com/engine/reference/builder/#scope.

Reply:
You are right that you need to define ARG in current scope. Documentation says where the scope starts, but not where it is ends.

In our case PLAT is [defined here](https://github.com/carsales/pyheif/pull/86/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R4) in `base` stage, and then it is used right here in `repaired` stage which inherits from `build-deps`, which inherits from `base`.

I can't find any proof in documentation how ARGs are inherited, but it works with ARG in the `base` stage and doesn't;t work otherwise.

By the way, what do you mean by "other builders"?

Reply:
I use podman on mac, which seems to be stricter about the scopes, not allowing args to be inherited.
Interestingly it's the first time I've come across a difference between them.

Reply:
Ok, added ARG to all related stages